docs/integrations/Instructions.md

```diff
@@ -0,0 +1,160 @@
+allowed. Additionally, short hand forms are provided to minimize representations
+on connected networks.
+
+Instructions are interpreted according to their non-Serai network. Addresses
+have no validation performed, beyond being a valid enum entry (when applicable)
+of the correct length, unless otherwise noted. If the processor is instructed to
+act on invalid data, or send to itself, it will drop the entire instruction.
+
+### Serialization
+
+  - Numbers are exclusively unsigned and encoded as compact integers under
+SCALE. If omitted, `0`.
+  - Enums are prefixed by an ordinal byte of their type, followed by their
+actual values.
+  - Vectors are prefixed by their length. If omitted, `vec![]`.
+  - Instruction fields are numbered and sequentially encoded, each prefixed by
+an ordinal byte. All other fields are sequentially encoded with no markers.
+
+Certain fields may be omitted depending on the network in question.
+
+### In Instructions
+
+  - `origin` (Address): Address from the network of origin which sent funds in.
+  - `target` (Address): The ink! contract to transfer the incoming funds to.
+  - `data`   (Vec<u8>): The data to call the target with.
+
+Networks may automatically provide `origin`. If they do, the instruction may
+still provide `origin`, overriding the automatically provided value. If no
+`origin` is provided, the instruction is dropped.
+
+Upon receiving funds, the respective Serai Asset contract is called, minting the
+appropriate amount of coins, and transferring them to the specified target with
+the attached data.
+
+If the transaction fails, funds are scheduled to be returned to `origin`.
+
+### Out Instructions
+
+  - `destination` (Enum { Native(Address), Serai(Address) }): Address to receive
+funds to.
+  - `data`        (Vec<u8>):                                  The data to call
+the target with.
+
+If the network is Serai, this is a transfer. Else, it's a withdrawal to the
+specified address with the specified data. Asset contracts perform no validation
+on these fields.
+
+### Shorthand
+
+All In Instructions are encoded as Shorthand. Shorthand is an enum which expands
+to an In Instruction.
+
+##### Raw
+
+Raw Shorthand encodes a raw In Instruction with no further processing.
+
+##### Swap
+
+  - `origin`  (Option<Address>): In Instruction's `origin`.
+  - `coin`    (Coin):            Coin to swap funds for.
+  - `minimum` (Amount):          Minimum amount of `coin` to receive.
+  - `out`     (Out Instruction): Final destination for funds.
+
+which expands to:
+
+```
+In Instruction {
+  origin,
+  target: Router,
+  data:   swap(Incoming Asset, out, minimum)
+}
+```
+
+where `swap` is a function which:
+
+  1) Swaps the incoming funds for SRI.
+  2) Swaps the SRI for `coin`.
+  3) Checks the amount of `coin` received is greater than `minimum`.
+  4) Executes `out` with the amount of `coin` received.
+
+For a Bitcoin to Monero swap, Swap Shorthand is expected to generally take:
```

Comment:
This may be better left to the PR comments, yet this does have to be less than 80 bytes. This math shows it is.